Container runtimes

nimrodoron opened the pull request:
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```improvement operator

```

danielfoehrKn committed:
adding health check library documentation

swilen-iwanow committed:
Introduce ShootState Sync controller

Commit:
Fixed labels in etcd resources to help in grouping

rfranzke committed:
Adapt all components to `core.gardener.cloud/v1beta1` API group

rfranzke committed:
Bump metrics-server version from `v0.3.3` to `v0.3.6`

See https://github.com/kubernetes/kubernetes/pull/83015

rfranzke committed:
Don't use deprecated hyperkube image anymore for 1.17+ clusters

See https://github.com/kubernetes/kubernetes/issues/81760 and https://github.com/kubernetes/kubernetes/pull/83454

rfranzke committed:
Add v1.17 to supported K8s version docs

rfranzke committed:
Update deprecated example CloudProfile and Shoot resources

shreyas-s-rao committed:
Add Grafana dashboard for ETCD Backups

Signed-off-by: Shreyas Rao <shreyas.sriganesh.rao@sap.com>

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Add runtime to shoot/worker spec

rfranzke committed:
Keep using hyperkube for binaries on worker nodes

We cannot just download binaries from `storage.googleapis.com` as this
would break private cloud and on-prem deployments where this location is
not accessible. Also, for these scenarios, we don't have a place where
to put the binaries on the worker nodes, and we probably don't want to
introduce this new dependency. One suitable solution would be to have
a container image like `hyperkube` that contains all the necessary
binaries, and during node-bootstrapping these binaries can be copied
over to the file system of the VM. As the Gardener project does not yet
have such a container we keep using `hyperkube` for now until we can
replace it with one built by us.

nimrodoronsap committed:
Fixes

nimrodoronsap committed:
Fixes

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
revert generated content

nimrodoronsap committed:
revert generated content

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Add runtime to shoot/worker spec

nimrodoronsap committed:
Fix type

nimrodoronsap committed:
Merge remote-tracking branch 'origin/container-runtimes' into container-runtimes

# Conflicts:
#	pkg/operation/shoot/types.go

nimrodoronsap committed:
Fix conversion

nimrodoronsap committed:
Add creation of container runtime reousrces during shoot reconciliation

nimrodoronsap committed:
Add annotation to container runtime supported workers

nimrodoronsap committed:
Unmark rest of container runtime logic

nimrodoronsap committed:
Fix shoot annotation for container runtimes

nimrodoronsap committed:
dummy

nimrodoronsap committed:
undo dummy

nimrodoronsap committed:
Fix shoot reconcile flow and delete

nimrodoronsap committed:
Add container runtime crd

ialidzhikov committed:
Make hvpa CRD v1.11 compatible

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

ialidzhikov committed:
./hack/generate-code for the new year

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

vpnachev committed:
Merge pull request #1777 from ialidzhikov/fix/hvpa-crd

Make hvpa CRD v1.11 compatible

rfranzke committed:
Merge pull request #1771 from gardener/feature/kubernetes-1.17

Support for Kubernetes v1.17

rfranzke committed:
Merge pull request #1778 from ialidzhikov/flake/new-year

./hack/generate-code for the new year

gardener-robot-ci-1 committed:
Upgrade github.com/gardener/vpn

from 0.16.0 to 0.17.0

liorokman commented on charts/seed-bootstrap/templates/extensions/crd-containerruntime.yaml:
```diff
@@ -0,0 +1,122 @@
+    singular: containerruntime
+    kind: ContainerRuntime
+    shortNames:
+      - cre
```
Is this really needed? `ContainerRuntime` is esoteric enough that there's no point in creating a short name for it.

nimrodoron replied:
Removed

liorokman reviewed (approved):
This just adds the new extension type, right?

rfranzke committed:
Merge pull request #1779 from gardener/ci-zqxubesch

[ci:component:github.com/gardener/vpn:0.16.0->0.17.0]

rfranzke committed:
Make node network optional for seeds and shoots

nimrodoron commented:
Gardener leftovers:
- Validation webhook
- Containerd unit service (log rotate and monitor)

nimrodoronsap committed:
Remove crd short name

rfranzke committed:
Merge pull request #1747 from georgekuruvillak/fix/fix-labels-in-etcd-resources

Fixed labels in etcd resources to help in grouping

timuthy committed:
Renounce hard-coded Kind user

Commit:
Merge pull request #1780 from timuthy/enhancement.dev-kind

Renounce hard-coded Kind user

Commit:
Merge pull request #1763 from rfranzke/feature/corev1beta1

Adapt all components to `core.gardener.cloud/v1beta1` API group

zanetworker committed:
Add documentation for the network extension

rfranzke committed:
Allow dynamic node networks generated by infra extensions

rfranzke committed:
Merge pull request #1775 from zanetworker/docs/network-extension

Add documentation for the Network resource

danielfoehrKn committed:
Removing mcm dependency from project & adjusting care controller to new extension health checks

timuthy committed:
Delete Kube-Apiserver service when hibernating

Commit:
Add ConditionBuilder helper functions

This allows for fluent update of condition and checks to ensure that
correct LastUpdateTime and LastTransitionTime are set only when the
condition has been modfied.

It also moves the metav1.Now function provided inside of the struct
making it easier for tests and without any global state modifications.

Commit:
Seed extension check controller status fix

It now updates the status only when there is an actual change in it.
Health checks on the controller installations (changes in their `.status`)
no longer cause the condition `ExtensionsReady` to have it's
`LastUpdateTime` constantly modified when there are no changes in
`Reason` and `Message` fields.

EmoinLanyu committed:
Add gardener-apiserver encryption config in gardener runtime charts

rfranzke committed:
Merge pull request #1786 from danielfoehrKn/healthckeck-clean-care-controller

Cleaning  care controller in gardenlet

rfranzke committed:
Merge pull request #1682 from danielfoehrKn/update-health-check-docs

adding health check library documentation

rfranzke committed:
Merge pull request #1787 from mvladev/rework-seed-extension-control

Rework seed extension control

nimrodoronsap committed:
Fix crd

rfranzke committed:
Merge pull request #1790 from EmoinLanyu/encrypt

Add gardener-apiserver encryption config in gardener runtime charts

rfranzke committed:
Merge pull request #1791 from timuthy/enhancement.hibernation

Delete Kube-Apiserver service when hibernating

Commit:
Merge pull request #1782 from rfranzke/feature/optional-nodes-cidr

Make nodes CIDR optional and allow it to be dynamic

rfranzke committed:
Scheduler uses shoot default networks during scheduling

Commit:
Merge pull request #1793 from gardener/rc

Scheduler uses shoot default networks during scheduling

nimrodoronsap committed:
Add registration to container runtime crd

timuthy committed:
Fix shoot deletion flow

ialidzhikov committed:
Add scheduler test for Seed .spec.networks.shootDefaults

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

rfranzke committed:
Merge pull request #1795 from timuthy/fix.hibernation

Fix shoot deletion flow

rfranzke committed:
Merge pull request #1798 from ialidzhikov/enh/scehduler-test

Add scheduler test for Seed .spec.networks.shootDefaults

vlvasilev committed:
Add GardenerOperationRestore

swilen-iwanow committed:
Adding functionality for turning off federated controllers

Commit:
Fix dashboard test for new dashboard 2.0

rfranzke committed:
Merge pull request #1801 from schrodit/fix-dahboard-test

Fix dashboard test for new 2.0 dashboard

rfranzke committed:
Merge pull request #1800 from vlvasilev/add_restore_annotation_constatnt

Add GardenerOperationRestore constant

rfranzke committed:
Remove provider-specific class types from autoscaler RBAC (#1803)

rfranzke committed:
Remove machineclass constant from Gardener APIs (#1804)

rfranzke committed:
Remove hard-coded calico fluentd config

rfranzke committed:
Merge pull request #1731 from swilen-iwanow/feature_sync_controller

Control Plane Migration ShootState Sync controller

rfranzke committed:
Merge pull request #1805 from rfranzke/cleanup/calico-logging-config

Remove hard-coded calico fluentd config

vpnachev committed:
Allow deletion of shoots running on tainted seeds

gardener-robot-ci-1 committed:
Release v0.34.0

Commit:
Prepare Next Dev Cycle v0.35.0-dev

rfranzke committed:
Remove MCM artifacts after removal in g/g (#1806)

nimrodoron committed:
Merge pull request #6 from gardener/master

merge

rfranzke committed:
Merge pull request #1794 from shreyas-s-rao/etcdbr-dashboard

Add Grafana dashboard for ETCD Backups

zanetworker committed:
fix warning messages appearing during coredns import

rfranzke committed:
Merge pull request #1808 from vpnachev/fix-validation/ignore-taints-for-existing-shoots

Allow update/deletion of shoots already running on tainted seeds

rfranzke committed:
Merge pull request #1815 from zanetworker/fix/warnings-coredns

Fix warning messages appearing during coredns import

jia-jerry committed:
Ensure value to be string in shoot-info chart (#1819)

nimrodoron committed:
Merge pull request #8 from gardener/master

merge

nimrodoronsap committed:
Change to garden code beta
